integration-tests/redis_fetch_manager_test.go

```diff
@@ -0,0 +1,58 @@
+//go:build integration
+
+package integration_tests
+
+import (
+	"context"
+	"github.com/nexodus-io/nexodus/internal/handlers/fetchmgr"
+	"github.com/nexodus-io/nexodus/internal/handlers/fetchmgr/redisfm"
+	"github.com/nexodus-io/nexodus/internal/handlers/fetchmgr/tests"
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+func TestFetchManager(t *testing.T) {
+
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "kubectl", "-n", "nexodus", "port-forward", "redis-0", "6379:6379")
+	cmd.Stdout = io.Discard
+	cmd.Stderr = os.Stderr
+	err := cmd.Start()
+	require.NoError(t, err)
+
+	logger, _ := zap.NewProductionConfig().Build()
+
+	client := redis.NewClient(&redis.Options{
+		Addr: "127.0.0.1:6379",
+	})
+	_, _ = client.Del(context.Background(), "key1").Result()
+	_ = client.Close()
```

Comment:
so the 2nd time you run the test, the redis "key1" will have items in it.  This is to delete those IF they exist.

Reply:
Something about this isn't working as intended. The test always passes for me the first time I run it, but fails every time after. If I restart redis, it will pass the first run again.

I also changed the code to make the test use a different cache key each time it runs and that makes it pass every time.

From some debug logging, it looks like during the failing runs, redis returns an empty list for `key1`. Is this code deleting the contents, but not the key?

Reply:
Think I got it fixed now, issue was that we did not wait long enough for the port-forward to be up and running, and the initial clean up to the redis keys was failing.